packages/nimble-components/src/wafer-map/tests/wafer-map.stories.ts

```diff
@@ -0,0 +1,213 @@
+import { html } from '@microsoft/fast-element';
+import type { Meta, StoryObj } from '@storybook/html';
+import { createUserSelectedThemeStory } from '../../utilities/tests/storybook';
+import { bodyFont } from '../../theme-provider/design-tokens';
+
+import '../../all-components';
+import type { WaferMapDie, WaferMapColorsScale } from '../types';
+import {
+    WaferMapQuadrant,
+    WaferMapOrientation,
+    WaferMapColorsScaleMode
+} from '../types';
+import {
+    highLightedValueSets,
+    wafermapDieSets,
+    waferMapColorsScaleSets
+} from './sets';
+
+interface WaferMapArgs {
+    colorscale: WaferMapColorsScale;
+    colorscalemode: WaferMapColorsScaleMode;
+    dielabelshidden: boolean;
+    dielabelsuffix: string;
+    dies: string;
+    highlightedvalues: string;
+    maxcharacters: number;
+    orientation: WaferMapOrientation;
+    quadrant: WaferMapQuadrant;
+}
+
+const metadata: Meta<WaferMapArgs> = {
+    title: 'WaferMap',
+    parameters: {
+        docs: {
+            description: {
+                component:
+                    'A wafer map is a component for visualizing data from the manufacture of semiconductor wafers. Each die on the wafer can show numerical information and be colored to indicate information about that die.'
+            }
+        }
+    },
+    render: createUserSelectedThemeStory(html`
+        <div id="usage-warning">
+            WARNING - The wafermap is still in development and considered
+            experimental. It is not recommended for application use.
+        </div>
+        <nimble-wafer-map
+            colorsScaleMode="${x => x.colorscalemode}"
+            dieLabelsHidden="${x => x.dielabelshidden}"
+            dieLabelsSuffix="${x => x.dielabelsuffix}"
+            maxCharacters="${x => x.maxcharacters}"
+            orientation="${x => x.orientation}"
+            quadrant="${x => x.quadrant}"
+            :colorScale="${x => x.colorscale}"
+            :dies=${x => {
+        let returnedValue: WaferMapDie[] | undefined;
+        switch (x.dies) {
+            case 'set1':
+                returnedValue = wafermapDieSets[0];
+                break;
+            case 'set2':
+                returnedValue = wafermapDieSets[1];
+                break;
+            case 'set3':
+                returnedValue = wafermapDieSets[2];
+                break;
+            case 'set4':
+                returnedValue = wafermapDieSets[2];
+                break;
+            default:
+                returnedValue = undefined;
+                break;
+        }
+        return returnedValue;
+    }}
+            :highlightedValues="${x => {
```

Comment:
I think here maybe you need to have 4 cases because in the "sets.ts" file you have 4 examples.